src/Servers/IIS/IIS/src/Core/IO/AsyncIOEngine.cs

```diff
@@ -34,7 +36,54 @@ public ValueTask<int> ReadAsync(Memory<byte> memory)
         return new ValueTask<int>(read, 0);
     }
 
-    public ValueTask<int> WriteAsync(ReadOnlySequence<byte> data)
+    public async ValueTask<int> WriteAsync(ReadOnlySequence<byte> data)
+    {
+        if (data.IsSingleSegment)
+        {
+            return await WriteImplAsync(data);
+        }
+        else
+        {
+            /*
+             * In case the number of chunks is bigger than responseMaxChunks we need to make multiple calls 
+             * to the native api https://docs.microsoft.com/en-us/iis/web-development-reference/native-code-api-reference/ihttpresponse-writeentitychunks-method  
+             * Despite the documentation states that feeding the function with more than 65535 chunks will cause the function to throw an exception, 
+             * it actually seems that 65534 is the maximum number of chunks allowed.
+             * Also, there seems to be a problem when slicing a ReadOnlySequence on segment borders tracked here https://github.com/dotnet/runtime/issues/67607
+             * That's why responseMaxChunks is set to be 65533. 
+             */
+
```

Comment:
nit: empty line after the comment and between length/result declarations on line 59 would be a little more compact

Reply:
Sure, will do.